File: aoc/2019/day1.py

with open('day1.txt', 'r') as f:
  lines = f.read().strip().split('\n')
  nums = [int(s) for s in lines]

def get_fuel(n):
  return int(n/3) - 2

fuels = [get_fuel(n) for n in nums]
print('part 1 ', sum(fuels))

corrected_fuels = []
for payload in nums:
  fuel = get_fuel(payload)
  total = 0
  while fuel > 0:
    total += fuel
    fuel = get_fuel(fuel)
  corrected_fuels.append(total)
print('part 2 ', sum(corrected_fuels))

File: aoc/2018/day1.js
const fs = require('fs')

const lines = fs.readFileSync('1.txt', 'utf-8').split('\n').map(i => parseInt(i))
let sum = 0

lines.forEach(i => sum += i)
const touched = {}

let i = 0
let value = 0
while (!touched[value]) {
  touched[value] = true
  value += lines[i % lines.length]
  i++
}

console.log(value)